### ISSUE TITLE
When using yolov8s to train, the weights of 8s are loaded. Why are the weights of yolov8n downloaded when training starts?

### ISSUE THREAD

**Issue by S-7-12** [ultralytics/ultralytics#14201]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When using yolov8s to train, the weights of 8s are loaded. Why are the weights of yolov8n downloaded when training starts?
My code is as follows:
```python
model = YOLO('ultralytics/cfg/models/v8/yolov8s.yaml')
model.load('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt') # loading pretrain 
```
The console output is as follows:
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v8.1.0/yolov8n.pt to 'yolov8n.pt'...
There is no error in the path to the weights file。

**Comment by ambitious-octopus** [comment]
Hey @S-7-12, please try:
```python
model = YOLO('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt')
```

**Comment by S-7-12** [comment]
> Hey @S-7-12, please try:
> 
> ```python
> model = YOLO('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt')
> ```

Same as the above output, no difference, still using yolo8n

**Comment by ambitious-octopus** [comment]
Hey @S-7-12 are you sure this path ('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt') is valid?

**Comment by 3045469829** [comment]
I have the same problem.

**Comment by S-7-12** [comment]
> Hey @S-7-12 are you sure this path ('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt') is valid?

The reason why I set it to an absolute path is because I am afraid that there will be problems with the path, so I am sure it works. And the weights were successfully loaded at the beginning of training

**Comment by glenn-jocher** [comment]
Hi @S-7-12-12,

Thank you for confirming the path. It seems like the issue might be related to how the model is being initialized and loaded. Let's try a slightly different approach to ensure the correct weights are being used:

1. **Directly Load the Pretrained Model**: Instead of building the model from the YAML file and then loading the weights, you can directly load the pretrained model using the `.pt` file.

```python
from ultralytics import YOLO

# Directly load the pretrained YOLOv8s model
model = YOLO('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt')

# Train the model
results = model.train(data='path/to/your/data.yaml', epochs=100, imgsz=640)
```

2. **Check for AMP Settings**: The message about AMP (Automatic Mixed Precision) checks with YOLOv8n suggests that the AMP settings might be triggering the download of YOLOv8n weights. Ensure that your environment is correctly set up for AMP if you intend to use it.

3. **Verify Package Versions**: Ensure you are using the latest version of the Ultralytics package. Sometimes, issues are resolved in newer releases.

```bash
pip install --upgrade ultralytics
```

4. **Reproducible Example**: If the issue persists, providing a minimum reproducible example can help us diagnose the problem more effectively. You can refer to our guide on creating a reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

If you continue to experience issues, please share more details about your environment and any additional console output. This will help us assist you better.

**Comment by Ahzyuan** [comment]
Encounter the same problem, and just set `amp=False` to disable amp in training setting, and it works.

**Comment by yang-fei** [comment]
> Hi @S-7-12-12,
> 
> Thank you for confirming the path. It seems like the issue might be related to how the model is being initialized and loaded. Let's try a slightly different approach to ensure the correct weights are being used:
> 
> 1. **Directly Load the Pretrained Model**: Instead of building the model from the YAML file and then loading the weights, you can directly load the pretrained model using the `.pt` file.
> 
> ```python
> from ultralytics import YOLO
> 
> # Directly load the pretrained YOLOv8s model
> model = YOLO('/home/nlp/swx/pycharm/ultralytics-main/yolov8s.pt')
> 
> # Train the model
> results = model.train(data='path/to/your/data.yaml', epochs=100, imgsz=640)
> ```
> 
> 2. **Check for AMP Settings**: The message about AMP (Automatic Mixed Precision) checks with YOLOv8n suggests that the AMP settings might be triggering the download of YOLOv8n weights. Ensure that your environment is correctly set up for AMP if you intend to use it.
> 3. **Verify Package Versions**: Ensure you are using the latest version of the Ultralytics package. Sometimes, issues are resolved in newer releases.
> 
> ```shell
> pip install --upgrade ultralytics
> ```
> 
> 4. **Reproducible Example**: If the issue persists, providing a minimum reproducible example can help us diagnose the problem more effectively. You can refer to our guide on creating a reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).
> 
> If you continue to experience issues, please share more details about your environment and any additional console output. This will help us assist you better.

In my cases, the message are from the AMP settings. Is it OK to use AMP for yolov8s model? So could we just ignore the message?
"AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅"

**Comment by SheffeyG** [comment]
I noticed the [checks](https://github.com/ultralytics/ultralytics/blob/edca88d1b3c9d387c2c316ec5e7817d2ee839014/ultralytics/utils/checks.py#L663) would load yolov8n.pt automatically, and that's why it downloads yolov8n.pt always if there's no yolov8n.pt file in project root directory, and they do that just for comparing the results between FP32 and AMP if I guess right, so the training model is which you just specified not that yolov8n.

**Comment by pderrenger** [comment]
You're absolutely right. The AMP checks do load `yolov8n.pt` to compare FP32 and AMP results, but this doesn't affect your specified training model. If you want to avoid this download, you can disable AMP by setting `amp=False` in your training settings. This way, your training will proceed with the model you specified without downloading additional weights.
